### ISSUE TITLE
Provide pre-build docker images on Docker Hub

### ISSUE THREAD

**Issue by simonla82** [OpenSlides/OpenSlides#5819]
**Is your feature request related to a problem? Please describe.**
When setting up a new OpenSlides instance I need to clone the git repo, check out the version I want, set various variables and build all the images (which can take quiet a long time). All in all this is an appropriate process for developers but as a **user** I do not need the source code at all and do not want to build the Docker images myself (in particular this contradicts the concept of Docker to provide containers containing **all** the needed dependencies imo)

**Describe the solution you'd like**
To resolve this problem one could:
- Build Docker images for each OpenSlides Version and push the accordingly tagged images to [Docker Hub](https://hub.docker.com) for all supported architectures
- Provide a sample `docker-compose.yml` which uses the built docker images from Docker Hub 
- Provide a sample `.env` file parsed by the `docker-compose.yml` file containing the configuration
- Maybe some Docker secret related stuff but I don't really have a clue of docker secrets so this would be open to discussion

To start a new OpenSlides instance I would then solely need to download the `docker-compose.yml` file, tweak the variables in the `.env` file (especially one could pass the OpenSlide version via the `.env` file) and fire up the docker-compose stack.

The build and push of the docker images could be automated via a [Github Action](https://github.com/marketplace/actions/build-and-push-docker-images) running on tags solely.

**Comment by gsiv** [comment]
Thank you for your detailed suggestion.

Since I've been addressed here, I don't think we are interested in building,
maintaining, and supporting official Docker images but maybe something could be
arranged through community participation?

**Comment by simonla82** [comment]
Thank your for your quick answer. Maybe I'm simplifying a bit dut aren't you already maintaining the docker images since the main deployment method is docker-compose? Since the varialbe `DEFAULT_DOCKER_REGISTRY` already exists in the `.env` file I think there would not be any changes to the compose file necessary.
So imo the only step necessary would be to create a Github action (instead of the `build.sh`) as shown [here](https://github.com/marketplace/actions/build-and-push-docker-images) to build and push the images for all newly created tags (for all supported architectures).

**Comment by FinnStutzenstein** [comment]
Yes and no. First, this must be maintained and some docs about environment variables are needed. Second, OpenSlides itself consists of multiple required containers, that must work together. It is nit possible to have one container to do everything. At least there, you need a docker-stack.yml (or some other orchestrational tool) which requires to use this repo unless the files are provided somewhere else. I guess that I missed some points, too.

If someone would do this, it would be nice.

**Comment by simonla82** [comment]
1. Yeah that's right.
2. Ok, I see your point. My proposal has not been to create one huge openslides image containing all the dependencies. I think two solutions two this problem exist:
One could create multiple docker hub repos (e.g. `openslides/openslides-client`, `openslides/openslides-server`, ...) containing the corresponding tags for the images. 
If one would want to avoid creating multiple repos the other option would be to tag the images accordingly (e.g. `openslides/openslides:server-latest`, `openslides/openslides:client-3.3`, ...) but I think that would miss the point of Docker Hub a bit. 
So I am with you that a docker-compose.yml (or similar) is needed as a orchestration tool. 

So my proposal would be to publish the setup guide (with an example docker-compose file) at docker hub (as done e.g. [here](https://hub.docker.com/_/joomla)) or provide a link to the repos Readme containing an example docker file.

But yes, probably some things would get a bit more complicated than on a first thought...

Personally I do not really have the time to look into this at the moment and I it's hard to put those changes into a pull request since it also requires changes in the openslides docker repo etc.

**Comment by FinnStutzenstein** [comment]
> One could create multiple docker hub repos

I was already thinking of it since one repo misses the point as you said. The way you proposeed is also my favorite one, but we currently do not have the time to do it ourselfs (also since our personal setups do not use dockerhub for our issues). So if someone wants to do it, it would be appreciated.
